### PR TITLE
[W6.2b][W12-1] Lim Jit Wei

### PR DESCRIPTION
**Pull request by jitwei98**
1.  Add a method isMutating() to the Command class
2. Modify Logic.java to save to storage only when data is mutated

### CODE REVIEW

File: src/seedu/addressbook/commands/Command.java [hunk 1]
```diff
@@ -46,6 +46,11 @@ public CommandResult execute(){
     //Note: it is better to make the execute() method abstract, by replacing the above method with the line below:
     //public abstract CommandResult execute();
 
+    /**
+     * Returns true if the command mutate the data.
```
Comment (okkhoy) [inline src/seedu/addressbook/commands/Command.java:50]: 👍 for following the correct format for the header comment!

File: src/seedu/addressbook/logic/Logic.java [hunk 2]
```diff
@@ -85,7 +85,9 @@ public CommandResult execute(String userCommandText) throws Exception {
     private CommandResult execute(Command command) throws Exception {
         command.setData(addressBook, lastShownList);
         CommandResult result = command.execute();
-        storage.save(addressBook);
+        if (command.isMutating()) {
```
Comment (okkhoy) [inline src/seedu/addressbook/logic/Logic.java:88]: You need to update the header comment for this `execute` method as the behavior has changed due to the way data is saved now!

**Review by okkhoy — COMMENTED** [review]
Please note that going forward you need to update the tests as well when you make any behaviour changes.

**Comment by okkhoy** [comment]
@jitwei98 
Some comments added. Please take note and close the pull request.

**Comment by jitwei98** [comment]
Noted with thanks!

**Pull request closed without merging**